skbio/stats/composition.py

```diff
@@ -608,10 +611,266 @@ def centralize(mat):
+    b6    2
+    b7    2
+    Name: W, dtype: int64
+    >>> results['reject']
+    b1     True
+    b2     True
+    b3    False
+    b4    False
+    b5    False
+    b6    False
+    b7    False
+    Name: reject, dtype: bool
+    """
+    if len(table) != len(categories):
+        raise ValueError('The number of samples in mat needs' +
+                         'to be the same as the number of samples' +
+                         'in cats')
+    if (multiple_comparisons_correction not in ['holm-bonferroni', None]):
+        raise NotImplementedError('%s has not been implemented'
+                                  % multiple_comparisons_correction)
+
+    mat = _check_composition(table, ignore_zeros=False)
+    cats = pd.Series(categories)
+
+    mat.sort_index(inplace=True)
+    cats = cats.sort_index()
+    labs = mat.columns
+
+    n_samp, n_feat = mat.shape
+
+    _logratio_mat = _log_compare(mat.values, cats.values, significance_test)
+    logratio_mat = _logratio_mat + _logratio_mat.T
+
+    # Multiple comparisons
+    if multiple_comparisons_correction == 'holm-bonferroni':
+        logratio_mat = np.apply_along_axis(_holm_bonferroni,
+                                           1, logratio_mat)
+    np.fill_diagonal(logratio_mat, 1)
+    W = (logratio_mat < alpha).sum(axis=1)
+    c_start = W.max() / n_feat
+    if c_start < theta:
+        reject = np.zeros_like(W, dtype=bool)
+    else:
+        # Select appropriate cutoff
+        cutoff = c_start - np.linspace(0.05, 0.25, 5)
+        prop_cut = np.array([(W > n_feat*cut).mean() for cut in cutoff])
+        dels = np.abs(prop_cut - np.roll(prop_cut, -1))
+        dels[-1] = 0
+        if (dels[1] < tau) and (dels[2] < tau) and (dels[3] < tau):
+            nu = cutoff[1]
+        elif (dels[1] >= tau) and (dels[2] < tau) and (dels[3] < tau):
+            nu = cutoff[2]
+        elif (dels[2] >= tau) and (dels[3] < tau) and (dels[4] < tau):
+            nu = cutoff[3]
+        else:
+            nu = cutoff[4]
+        reject = (W >= nu*n_feat)
+    return pd.DataFrame({'W': pd.Series(W, index=labs),
+                         'reject': pd.Series(reject, index=labs)})
+
+
+def _check_composition(x, ignore_zeros=True):
+    """ Checks to make sure that composition meets the mininum criteria
+    Also casts composition into a pandas dataframe
+
+    Parameters
+    ----------
+    x : array_like or pd.DataFrame
+       Input composition matrix
+       where rows=samples and columns=features
+    ignore_zeros : bool
+       Specifies if compositions are allowed to have zeros
+
+    Returns
+    -------
+    pd.DataFrame
+       Validated composition matrix
+    """
+    if x.ndim > 2:
+        raise ValueError("Input matrix can only have two dimensions or less")
+    if isinstance(x, pd.DataFrame):
+        mat = closure(x)
+        samp_labs = x.index
+        feat_labs = x.columns
+    else:
+        mat = np.atleast_2d(x)
+        r, c = mat.shape
+        samp_labs = range(r)
```

Comment:
Can you use `future.builtins.range`? See our [py2/3 guide](http://scikit-bio.org/docs/latest/development/py3.html) for details.